### ISSUE TITLE
The request signature we calculated does not match the signature you provided. SigV4Auth vs CrtSigV4Auth 

### ISSUE THREAD

**Issue by JorisLA** [boto/botocore#2358]
**Describe the bug**
greengrass reset_deployment api raises an error

**Steps to reproduce**
The OS is Linux 64 bit, python3.7.5

boto3==1.17.54
botocore==1.20.54

Credentials comes from a user with all rights on greengrass and iot.
default_region is eu-west-1.
The aws_secret_access_key only has alphanumeric characters.

```
client = boto3.client(
    'greengrass',
    aws_access_key_id=config.aws_access_key_id,
    aws_secret_access_key=config.aws_secret_access_key,
    region_name=config.aws_default_region
)

reset_response = client.reset_deployments(Force=True, GroupId=group_id)
```

**Debug logs**
```
botocore.exceptions.ClientError: An error occurred (InvalidSignatureException) when calling the ResetDeployments operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
          
The Canonical String for this request should have been
'POST
/greengrass/groups/88d82b49-65a0-4b0a-a341-e238e22f4e86/deployments/%24reset
           
host:greengrass.eu-west-1.amazonaws.com
x-amz-date:20210420T131121Z

host;x-amz-date
e2c561fddf9afd1986904a4fe67cda75395efe3f30ba6cfb47bc888c7cd5de42'

The String-to-Sign should have been
'AWS4-HMAC-SHA256
20210420T131121Z
20210420/eu-west-1/greengrass/aws4_request
ba7b731ab265895ee07216aaf8ca67e816cdb377e8dd13cf8d4581cf451147b8'
```
It seems the error comes just after 1.20.52 botocore version.

When using CrtSigV4Auth class the error appears but with SigV4Auth it works fine.
To switch from one to the other I export BOTO_DISABLE_CRT to true

PS: It also works fine with aws cli

```
aws --version
aws-cli/1.19.54 Python/2.7.18 Linux/5.4.0-48-generic botocore/1.20.54
aws greengrass reset-deployments \
    --group-id "ad9d7e24-17dc-4d81-ba66-3619bb06645c" \
    --force
```

**Comment by nateprewitt** [comment]
Thanks for opening a new ticket @JorisLA! Could you also include the version of `awscrt` you had installed and where it was originally installed from? i.e. An AWS IoT SDK, Amazon-Transcribe-Streaming-SDK, `botocore[crt]`, or somewhere else?

**Comment by JorisLA** [comment]
awscrt==0.11.8 installed from awsiotsdk==1.5.12

**Comment by nateprewitt** [comment]
Thanks for the info! That helped us pinpoint the issue and I think we've got a good lead to getting things resolved. It appears there's currently a bug in how the CRT signs SigV4 for certain service configurations. We're going to work with the CRT team to see if we can get this fixed and I'll update the ticket when we have more.

**Comment by nateprewitt** [comment]
Hey @JorisLA,

Just wanted to provide an update. The CRT team has been able to fix the problem in `awscrt==0.11.15` which was released yesterday. We're in the process of getting that wired up in botocore, and it may take a bit longer for the IoT SDKs to catch up, but once that's finished, things should be working as expected. Thanks again for the report!

I'll close this out once we get the fix deployed in botocore unless you have any other questions.

**Comment by JorisLA** [comment]
Thanks for the update ! I don't have any other questions.

**Comment by nateprewitt** [comment]
This was resolved today in 1.20.71.
